File: Python/leetcode/rookie/01_删除排序数组重复.py


def removeDuplicates(nums):
    left = 0
    for right in range(1, len(nums)):
        if nums[left] != nums[right]:
            nums[left] = nums[right]
            left = left + 1
    return left + 1

a = removeDuplicates([1,1,2])
print(a)

if __name__ == 'main':
  print(removeDuplicates([1, 1, 2]))

File: leetcode/rookie/01_删除排序数组重复.js

var removeDuplicates = function(nums){
  // if(!nums.length){
  //   return 0
  // }
  // let left = 0
  // for (let right = 1; right < nums.length; right++) {
  //   if(nums[left] !== nums[right]){
  //     nums[++left] = nums[right]
  //   }
  //   console.log('left', left)
  // }
  nums = [...new Set(nums)]
  console.log(nums)
  return nums.length
}

console.log(removeDuplicates([1,1,2]))